changelog/next/features/4773--no-store-internal-events.md

```diff
@@ -0,0 +1,3 @@
+You can now disable the persistence of internally generated events like
+diagnostics and metrics by setting the option `no-store-internal-events` to
```

Comment:
While the name describes accurately what it does internally, this setting will be very hard to discover for a user who just wants to figure out how to disable metrics.

From that perspective, options like `store-metrics` or `metrics-sink=internal|none` would be easier to understand.

Reply:
Tobias and I discussed earlier that I will take this over, and alongside that I'll try to come up with a new name.